extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/VertxHybridPoolObjectMapperCustomizer.java

```diff
@@ -0,0 +1,14 @@
+package io.quarkus.jackson.runtime;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.jackson.ObjectMapperCustomizer;
+import io.vertx.core.json.jackson.HybridJacksonPool;
+
+public class VertxHybridPoolObjectMapperCustomizer implements ObjectMapperCustomizer {
+
+    @Override
+    public void customize(ObjectMapper objectMapper) {
+        objectMapper.getFactory().setRecyclerPool(HybridJacksonPool.getInstance());
```

Comment:
I guess, this method is only called once.
So, we can try to verify if the `HybridJacksonPool` is available; otherwise, do nothing.

Of course, it would be even better to do that at build time and pass the resulting boolean to a recorder.

Reply:
The `HybridJacksonPool` will be available if there is the vertx-core capability, which is the thing that I'm testing here https://github.com/quarkusio/quarkus/pull/38196/files#diff-6565aad037315afaf46e89a1d2d5255711b128a5ea228721b9e1b0e340b706a6R114

Reply:
I am actually wondering if we should only set this if the user has not set one

Reply:
> I am actually wondering if we should only set this if the user has not set one

This makes sense, but the only way I could check this is doing something like
```
if (objectMapper.getFactory()._getRecyclerPool() != JsonRecyclerPools.defaultPool()) {
    objectMapper.getFactory().setRecyclerPool(HybridJacksonPool.getInstance());
}
```
Here the problem is that the jackson convention is that methods having a name starting with a `_`, like that `_getRecyclerPool()`, have to be considered NOT part of the public API even if they are public for implementation reasons. In this case I believe it should be fine to use that method, but please let me know if this is ok for you.

Reply:
Done.